src/main/java/org/spongepowered/common/world/WorldManager.java

```diff
@@ -540,37 +539,35 @@ public static boolean unloadWorld(final WorldServer worldServer, final boolean c
             final WorldServerBridge mixinWorldServer = (WorldServerBridge) worldServer;
             final int dimensionId = mixinWorldServer.bridge$getDimensionId();
 
-            try {
```

Comment:
I was just having another think about this. This try/catch/finally should be reinstated. The change in order is okay, but if the save fails, we want to clean up gracefully - removing this means that it leave us in an inconsistent state. The catch should change to just `Exception` or even `Throwable`, I'd think.

Similarly, just been talking to @Zidane about this, we should also `try/catch (Exception)` shutting down the async lighting thread too, for similar reasons.

Reply:
Ok so I should just change `catch (MinecraftException e)` to `catch (Exception e)` and do the same around the async lighting shutdown ?

No need to keep a `finally` block if all exceptions are caught and no exception is rethrown. All lines after the try catch blocks will be executed in any case.

Reply:
> No need to keep a finally block if all exceptions are caught and no exception is rethrown. All lines after the try catch blocks will be executed in any case.

If some other `Error` is thrown, it might - though they tend not to be recoverable, in the case it is we should still clean up.

Reply:
If we stick to `Exception` you're right. Do you want to catch `Exception` or `Throwable` ? `Throwable` seems maybe a bit overkill to me.

Reply:
Exception is fine.